R/versioning.R

```diff
@@ -0,0 +1,97 @@
+#' Turn bucket versioning on or off, check status (default), or
+#' list both live and archived versions of objects in the bucket
+#' @param action "status", "enable", "disable", or "list"
+#' @param bucket gcs bucket
+#'
+#' @return versioned_objects list #only if "list" action
+#' @importFrom googleAuthR gar_api_generator
+#' @family managementAPI functions
+#' @export
+gcs_version_bucket <- function(bucket, action = "status") {
```

Comment:
This sets the default to status, self documents the other possible arguments and hard codes users to only be able to choose from the list, to avoid weird errors.
```suggestion
gcs_version_bucket <- function(bucket, action = c("status","enable","disable","list") {
  action <- match.arg(action)
```